cpp/Platform.Setters/Setter[TResult, TDecision].h

```diff
@@ -1,46 +1,47 @@
-﻿namespace Platform::Setters
-{
-    template <typename ...> class Setter;
-    template <typename TResult, typename TDecision> class Setter<TResult, TDecision> : public SetterBase<TResult>
-    {
-        private: TDecision _trueValue = 0;
-        private: TDecision _falseValue = 0;
-
-        public: Setter(TDecision trueValue, TDecision falseValue, TResult defaultValue)
-            : SetterBase(defaultValue)
-        {
-            _trueValue = trueValue;
-            _falseValue = falseValue;
-        }
-
-        public: Setter(TDecision trueValue, TDecision falseValue) : this(trueValue, falseValue, 0) { }
-
-        public: Setter(TResult defaultValue) : SetterBase(defaultValue) { }
-
-        public: Setter() { }
-
-        public: TDecision SetAndReturnTrue(TResult value)
-        {
-            _result = value;
-            return _trueValue;
-        }
-
-        public: TDecision SetAndReturnFalse(TResult value)
-        {
-            _result = value;
-            return _falseValue;
-        }
-
-        public: TDecision SetFirstAndReturnTrue(IList<TResult> &list)
-        {
-            _result = list[0];
-            return _trueValue;
-        }
-
-        public: TDecision SetFirstAndReturnFalse(IList<TResult> &list)
-        {
-            _result = list[0];
-            return _falseValue;
-        }
-    };
-}
+﻿namespace Platform::Setters
+{
+    template <typename ...> class Setter;
+    template <typename TResult, typename TDecision> class Setter<TResult, TDecision> : public SetterBase<TResult>
+    {
+        private: TDecision _trueValue = 0;
+        private: TDecision _falseValue = 0;
+
+        using base = SetterBase<TResult>;
+
+        public: Setter(TDecision trueValue, TDecision falseValue, TResult defaultValue)
+        {
+            _trueValue = trueValue;
+            _falseValue = falseValue;
+            base::_result = std::move(defaultValue);
+        }
+
+        public: Setter(TDecision trueValue, TDecision falseValue) : Setter(trueValue, falseValue, 0) { }
+
+        public: Setter(TResult defaultValue) : base(defaultValue) { }
+
+        public: Setter() { }
+
+        public: TDecision SetAndReturnTrue(TResult value)
+        {
+            base::_result = std::move(value);
+            return _trueValue;
+        }
+
+        public: TDecision SetAndReturnFalse(TResult value)
+        {
+            base::_result = std::move(value);
+            return _falseValue;
+        }
+
+        public: TDecision SetFirstAndReturnTrue(Platform::Interfaces::IArray<TResult> auto&& list)
+        {
+            base::_result = list[0];

```

Comment:
If you true plusist [maybe] replace to
```cpp
if constexpr (std::is_rvalue_reference_v<decltype(list)>)
{
    base::_result = std::move(list[0]);
}
else 
{
    base::_result = list[0];
}
```

Reply:
Can we use https://en.cppreference.com/w/cpp/concepts/movable here?

Reply:
I think we cannot move the item of the list here, it will be changed this way, and the list will be not able to be reused later.